torch/csrc/nnpack/NNPACK.cpp

```diff
@@ -0,0 +1,348 @@
+#include "NNPACK.h"
+
+#include "TH/TH.h"
+#include <stdlib.h>
+
+namespace torch {
+namespace nnpack {
+
+// Stolen from Caffe2
+static pthreadpool_t nnpack_threadpool_ = nullptr;
+
+pthreadpool_t nnpack_threadpool() {
+  if (nnpack_threadpool_ == nullptr) {
+    enum nnp_status nnpack_status = nnp_initialize();
+    if (nnpack_status != nnp_status_success) throw std::runtime_error("could not initialize NNPack");
+    nnpack_threadpool_ = pthreadpool_create(16);
```

Comment:
I doubt this call cannot fail, so please add error checking

Reply:
Will do.